test/hotspot/jtreg/serviceability/jvmti/events/NotifyFramePopStressTest/libNotifyFramePopStressTest.cpp

```diff
@@ -0,0 +1,199 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "jvmti.h"
+#include "jvmti_common.hpp"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static jvmtiEnv *jvmti = nullptr;
+static jvmtiCapabilities caps;
+static jvmtiEventCallbacks callbacks;
+static volatile jint pop_count = 0;
+static char* volatile last_notify_method;
+static volatile jboolean failed = JNI_FALSE;
+static jboolean seenMain = JNI_FALSE;
+
+static jint Agent_Initialize(JavaVM *jvm, char *options, void *reserved);
+
+JNIEXPORT
+jint JNICALL Agent_OnLoad(JavaVM *jvm, char *options, void *reserved) {
+  return Agent_Initialize(jvm, options, reserved);
+}
+
+JNIEXPORT
+jint JNICALL Agent_OnAttach(JavaVM *jvm, char *options, void *reserved) {
+  return Agent_Initialize(jvm, options, reserved);
+}
+
+JNIEXPORT
+jint JNICALL JNI_OnLoad(JavaVM *jvm, void *reserved) {
+  return JNI_VERSION_9;
+}
+
+static void JNICALL
+FramePop(jvmtiEnv *jvmti, JNIEnv *jni, jthread thread,
+         jmethodID method, jboolean wasPoppedByException) {
+  jvmtiError err;
+  char* expected_method = (char*)last_notify_method;
+  jclass cls = nullptr;
+  char* csig = nullptr;
+  char* name = nullptr;
+
+  pop_count++;
+
+  err = jvmti->GetMethodDeclaringClass(method, &cls);
+  check_jvmti_status(jni, err, "FramePop: Failed in JVMTI GetMethodDeclaringClass");
+
+  err = jvmti->GetClassSignature(cls, &csig, nullptr);
+  check_jvmti_status(jni, err, "FramePop: Failed in JVMTI GetClassSignature");
+
+  name = get_method_name(jvmti, jni, method);
+  LOG("FramePop(%d) event from method: %s %s\n", pop_count, csig, name);
+
+  if (strcmp(name, "main") != 0) { // ignore FRAME_POP for main that comes in as the test exits
+    if (strcmp(name, (char*)expected_method) != 0) {
+      LOG("ERROR: FramePop event is for wrong method: expected %s, got %s\n", last_notify_method, name);
+      failed = JNI_TRUE;
+      deallocate(jvmti, jni, csig);
+      deallocate(jvmti, jni, name);
+    }
+  }
+  deallocate(jvmti, jni, csig);
+  deallocate(jvmti, jni, name);
```

Comment:
on error `csig` and `name` are deallocated twice

Reply:
Agreed, thanks. Removed the deallocation on error.